src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy

```diff
@@ -369,24 +312,122 @@ final class OxfordNanoporeMeasurement {
         return this.measurementFolder.relativePath
     }
 
-    /*
-    Inner class that contains the logic for the metadata validation
-     */
-    private static class MetaData {
-
-        private static final SCHEMA = "/schemas/ont-metadata.schema.json"
-
-        static validateMetadata(Map metaData) throws ValidationException {
-            // Load schema
-            final def metaDataJson = new JSONObject(metaData)
-            final def schemaStream = OxfordNanoporeMeasurement.getResourceAsStream(SCHEMA)
-            final def rawSchema = new JSONObject(new JSONTokener(schemaStream))
-            final def jsonSchema = SchemaLoader.load(rawSchema)
-            // Validate against schema
-            jsonSchema.validate(metaDataJson)
+    private static class Metadata {
+        private static final Map<String, ?> SCHEMA = parseMetadataSchema()
+        private static final String LIBRARY_PREP_KIT_SCHEMA = "SQK-.*(?=:)"
+
+        private String adapter
+        private String asicTemp
+        private String deviceType
+        private String flowcellId
+        private String flowcellPosition
+        private String flowcellType
+        private String baseCaller
+        private String baseCallerVersion
+        private String libraryPreparationKit
+        private String machineHost
+        private String startDate
```

Comment:
can we make them public but final? so we dont need the getter and just provide immutable content?

Reply:
You will always use the same schema and parse it with the constructor

Reply:
This way we do not have to introduce a new constructor for every eventuality but can use the factory method. I'd say this is good as is.